src/client/nativeCommunication.ts

```diff
@@ -127,7 +110,73 @@ function slideshow(): void {
         }));
 }
 
+function getVideoSlots(): VideoSlot[] {
+    const videoSlots = document.querySelectorAll('.native-video');
+
+    if (!videoSlots) {
+        return [];
+    }
+
+    return Array.from(videoSlots).map(videoSlot => {
+        const slotPosition = videoSlot.getBoundingClientRect();
+        return new VideoSlot({
+            rect: getRect(slotPosition),
+            videoId: videoSlot.getAttribute('data-videoId') ?? '',
```

Comment:
I think I'd rather have the whole VideoSlot be absent, than having the some properties be filled with empty strings.
It will make the job easier for the native side as they'll have to ignore that video slot anyway.

Reply:
yes we make sure we have these fields server side but the browser API can still return `undefined`. It might be worth handling undefined again though.

Reply:
Made this a reduce so only slots with `videoId` will be returned